# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/VideoDecoder.webidl
# Commit: eb17a0f3efdf
# Full Hash: eb17a0f3efdf67d5c7098c4ec5dc8ee57ea0956c
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-29 15:31:45
# Regressor Bug: 1831451
# File Overlap Count: 2
# Description:
#   Bug 1831451 - Add a blank VideoDecoder interface r=webidl,padenot,edgar,smaug,saschanaz
#   
#   This patch introduces a VideoDecoder interface generated by running
#   `./mach webidl-example VideoDecoder` with necessary changes to make it
#   buildable.
# ==============================================================================

diff -r 7377bc0dc1f6 -r eb17a0f3efdf dom/webidl/VideoDecoder.webidl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/webidl/VideoDecoder.webidl	Thu Jun 29 04:15:00 2023 +0000
@@ -0,0 +1,71 @@
+/* -*- Mode: IDL; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * The origin of this IDL file is
+ * https://w3c.github.io/webcodecs/#videodecoder
+ */
+
+[Exposed=(Window,DedicatedWorker), SecureContext, Pref="dom.media.webcodecs.enabled"]
+interface VideoDecoder : EventTarget {
+  [Throws]
+  constructor(VideoDecoderInit init);
+
+  readonly attribute CodecState state;
+  readonly attribute unsigned long decodeQueueSize;
+  attribute EventHandler ondequeue;
+
+  [Throws]
+  undefined configure(VideoDecoderConfig config);
+  [Throws]
+  undefined decode(EncodedVideoChunk chunk);
+  [NewObject, Throws]
+  Promise<undefined> flush();
+  [Throws]
+  undefined reset();
+  [Throws]
+  undefined close();
+
+  [NewObject, Throws]
+  static Promise<VideoDecoderSupport> isConfigSupported(VideoDecoderConfig config);
+};
+
+dictionary VideoDecoderInit {
+  required VideoFrameOutputCallback output;
+  required WebCodecsErrorCallback error;
+};
+
+callback VideoFrameOutputCallback = undefined(VideoFrame output);
+
+dictionary VideoDecoderSupport {
+  boolean supported;
+  VideoDecoderConfig config;
+};
+
+dictionary VideoDecoderConfig {
+  required DOMString codec;
+  // Bug 1696216: Should be 1696216 [AllowShared] BufferSource description;
+  ([AllowShared] ArrayBufferView or [AllowShared] ArrayBuffer) description;
+  [EnforceRange] unsigned long codedWidth;
+  [EnforceRange] unsigned long codedHeight;
+  [EnforceRange] unsigned long displayAspectWidth;
+  [EnforceRange] unsigned long displayAspectHeight;
+  VideoColorSpaceInit colorSpace;
+  HardwareAcceleration hardwareAcceleration = "no-preference";
+  boolean optimizeForLatency;
+};
+
+enum HardwareAcceleration {
+  "no-preference",
+  "prefer-hardware",
+  "prefer-software",
+};
+
+enum CodecState {
+  "unconfigured",
+  "configured",
+  "closed"
+};
+
+callback WebCodecsErrorCallback = undefined(DOMException error);